harness/determined/core/_train.py

```diff
@@ -231,6 +240,9 @@ def report_validation_metrics(self, steps_completed: int, metrics: Dict[str, Any
             f"report_validation_metrics(steps_completed={steps_completed} metrics={metrics})"
         )
 
+    def upload_tb_profile(self, rank: int = 0) -> None:
+        logger.info(f"upload_tb_profile({rank})")
+
```

Comment:
For going into a public api, this is too specific to the current bug we are looking to fix.  The behavior as-written is:
- we automatically upload chief metrics to tensorboard
- you can upload profiling metrics manually 

I think a better setup would be:
- you upload the tensorboard paths you want manually on every rank
- optionally: we could have an "autotensorboard" mode which is exactly what we have on tip-of-master.